### ISSUE TITLE
Overall health status for server

### ISSUE THREAD

**Issue by cortopy** [hyperium/tonic#598]
### Version

```
│   └── tonic v0.4.1
├── tonic v0.4.1 (*)
├── tonic-health v0.3.0
│   └── tonic v0.4.1 (*)
│   └── tonic-build v0.4.1
└── tonic-build v0.4.1 (*)

```

### Platform

Linux laptop 5.4.108-1-MANJARO #1 SMP PREEMPT Wed Mar 24 18:24:35 UTC 2021 x86_64 GNU/Linux

### Crates

tonic-health

### Description

According to the [specs](https://github.com/grpc/grpc/blob/master/doc/health-checking.md):

> The server should use an empty string as the key for server's overall health status, so that a client not interested in a specific service can query the server's status with an empty request

I tried this request using bloomrpc and the proto spec from the [docs](https://github.com/grpc/grpc/blob/master/doc/health-checking.md):

```
{
   "service": ""
}
```

I expected to tell me server is ok

Instead, I got this reponse:
```
{
  "error": "5 NOT_FOUND: service not registered"
}
```

**Comment by davidpdrsn** [comment]
What does "overall server health" mean specifically? Is the server always healthy if it can respond to a health check request or should other checks be performed?

If the server is always considered healthy then tonic-health can probably just set that internally during construction. But if something more custom is required then it might be better for users to call [`health_reporter.set_serving_status("", ServingStatus::Serving)`](https://docs.rs/tonic-health/0.3.1/tonic_health/server/struct.HealthReporter.html#method.set_service_status) and writing code to manually update the health status.

The spec also contains this sentence:

> The server can just do exact matching of the service name without support of any kind of wildcard matching.

Which makes it sound like _not_ supporting querying on `""` is fine.

**Comment by LucioFranco** [comment]
This has been fixed in `0.6`
